pubsub.go

```diff
@@ -162,34 +183,49 @@ func (p *PubsubValueStore) Subscribe(key string) error {
 		return cmp > 0 || cmp == 0 && src == myID
 	})
 
-	sub, err := p.ps.Subscribe(topic)
+	ti, err := p.createTopicHandler(topic)
 	if err != nil {
 		return err
 	}
 
-	p.mx.Lock()
-	existingSub, bootstrapped := p.subs[key]
-	if existingSub != nil {
-		p.mx.Unlock()
-		sub.Cancel()
-		return nil
-	}
-
+	p.topics[key] = ti
 	ctx, cancel := context.WithCancel(p.ctx)
-	p.cancels[key] = cancel
+	ti.cancel = cancel
```

Comment:
kind of annoying that this isn't handled inside `createTopicHandler`, but it's easier to deal with the contexts this way